src/marqo/models.py

```diff
@@ -24,9 +26,134 @@ class SearchBody(BaseMarqoModel):
     scoreModifiers: Optional[Dict] = None
     modelAuth: Optional[Dict] = None
 
+
 class BulkSearchBody(SearchBody):
     index: str
 
 
 class BulkSearchQuery(BaseMarqoModel):
     queries: List[BulkSearchBody]
+
+
+index_defaults_deprecated_to_new_params_mapping = {
+    "inference_node_type": "inference_type",
+    "storage_node_type": "storage_class",
+    "inference_node_count": "number_of_inferences",
+    "storage_node_count": "number_of_shards",
+    "replicas_count": "number_of_replicas",
+}
+
+params_that_can_be_specified_with_settings_dict = ["wait_for_readiness", "settings_dict"]
+
+
+class CreateIndexSettings:
```

Comment:
you can actually subclass a Pydantic basemodel which provides some boilerplate funtionality. maybe you'd find it helpful? For example: https://github.com/marqo-ai/marqo/blob/mainline/src/marqo/tensor_search/models/add_docs_objects.py#L35